pkg/ccl/logictestccl/tests/local/local_test: TestCCLLogic_as_of failed

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/5807fa68-e785-4b1f-b8f6-18a7039fd7bb?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [c471a724f99a5add3ccefe07995251760d0e2212](https://github.com/cockroachdb/cockroach/commits/c471a724f99a5add3ccefe07995251760d0e2212):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of2310056590
    test_log_scope.go:76: use -show-logs to present logs inline
[13:34:44] setting distsql_workmem='25709B';
    logic.go:2717: 
         
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:63: 
        expected:
        (pq: cannot execute CREATE DATABASE in a read-only transaction|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[13:34:45] --- done: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 16 tests, 1 failures
    logic.go:4261: 
        /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: error while processing
    logic.go:4261: /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of2310056590
--- FAIL: TestCCLLogic_as_of (7.07s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48056

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/6c394723-95a6-43f2-9433-201aad48d5ed?testReportRun=3&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [cbb292c487ea7413772a5c3c349f93467264c190](https://github.com/cockroachdb/cockroach/commits/cbb292c487ea7413772a5c3c349f93467264c190):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of2362135831
    test_log_scope.go:76: use -show-logs to present logs inline
[12:45:43] setting distsql_workmem='28741B';
    logic.go:2717: 
         
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:63: 
        expected:
        (pq: cannot execute CREATE DATABASE in a read-only transaction|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[12:45:44] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 16 tests, 1 failures
    logic.go:4261: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: error while processing
    logic.go:4261: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of2362135831
--- FAIL: TestCCLLogic_as_of (5.68s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/7107b300-6edd-49c7-9c29-28843f964f80?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [c44b3134d23d6037e1d5d5e12461a6914aa588e3](https://github.com/cockroachdb/cockroach/commits/c44b3134d23d6037e1d5d5e12461a6914aa588e3):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of2203961303
    test_log_scope.go:76: use -show-logs to present logs inline
[12:46:06] setting distsql_workmem='53024B';
    logic.go:2717: 
         
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:63: 
        expected:
        (pq: cannot execute CREATE DATABASE in a read-only transaction|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[12:46:07] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 16 tests, 1 failures
    logic.go:4261: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: error while processing
    logic.go:4261: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: too many errors encountered, skipping the rest of the input
    panic.go:629: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of2203961303
--- FAIL: TestCCLLogic_as_of (5.07s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/3663ac57-b7ef-4d21-bc29-e847c4af7f1f?testReportRun=3&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [e527f6b34c009ed7b52777e854687fb5d6ce3378](https://github.com/cockroachdb/cockroach/commits/e527f6b34c009ed7b52777e854687fb5d6ce3378):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of1398399245
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:2717: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:63: 
        expected:
        (pq: cannot execute CREATE DATABASE in a read-only transaction|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[14:16:07] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 16 tests, 1 failures
    logic.go:4261: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: error while processing
    logic.go:4261: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: too many errors encountered, skipping the rest of the input
    panic.go:629: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of1398399245
--- FAIL: TestCCLLogic_as_of (6.17s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #142174 pkg/ccl/logictestccl/tests/local/local_test: TestCCLLogic_as_of failed [C-test-failure O-robot T-sql-foundations branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/logictestccl/tests/local/local_test.TestCCLLogic_as_of [failed](https://tanzanite.cluster.engflow.com/invocations/default/d91e0564-e9df-4c9c-b276-c23d2afc5caf?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on master @ [8c5af013aa725d4f441abeff282894ff22f04ba7](https://github.com/cockroachdb/cockroach/commits/8c5af013aa725d4f441abeff282894ff22f04ba7):


```
=== RUN   TestCCLLogic_as_of
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCCLLogic_as_of535505161
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:2717: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:63: 
        expected:
        (pq: cannot execute CREATE DATABASE in a read-only transaction|database \"\[1\]\" does not exist)
        got:
        pq: role/user "root" does not exist
[13:25:43] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of with config local: 16 tests, 1 failures
    logic.go:4261: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: error while processing
    logic.go:4261: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/as_of:66: too many errors encountered, skipping the rest of the input
    panic.go:629: -- test log scope end --
test logs left over in: outputs.zip/logTestCCLLogic_as_of535505161
--- FAIL: TestCCLLogic_as_of (4.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #142174 pkg/ccl/logictestccl/tests/local/local_test: TestCCLLogic_as_of failed [C-test-failure O-robot T-sql-foundations branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_as_of.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/logictestccl/testdata/logic_test/as_of
+++ b/pkg/ccl/logictestccl/testdata/logic_test/as_of
@@ -60,11 +60,17 @@ ROLLBACK
 statement error pq: AS OF SYSTEM TIME specified with READ WRITE mode
 BEGIN READ WRITE
 
+statement ok
+SET autocommit_before_ddl = false
+
 statement error (pq: cannot execute CREATE DATABASE in a read-only transaction|database \"\[1\]\" does not exist)
 BEGIN; CREATE DATABASE IF NOT EXISTS d2
 
 statement ok
 ROLLBACK
+
+statement ok
+RESET autocommit_before_ddl
 
 statement ok
 SET DEFAULT_TRANSACTION_USE_FOLLOWER_READS TO FALSE

